sdk/typespec/typespec_client_core/src/http/headers/mod.rs

```diff
@@ -146,13 +192,19 @@ impl Headers {
     }
 
     /// Add headers to the headers collection.
-    pub fn add<H>(&mut self, header: H)
+    ///
+    /// ## Errors
+    ///
+    /// The error this returns depends on the type `H`.
+    /// Many header types are infallible, and do not return
```

Comment:
Nit: for public docs, we should strive for good grammar.
```suggestion
    /// Many header types are infallible, and do not return.
```

Reply:
I actually just forgot to write the rest of that section 😅